src/main/java/org/openrewrite/staticanalysis/MoveFieldsToTopOfClass.java

```diff
@@ -0,0 +1,136 @@
+               "grouping field declarations together at the top of the class. Comments associated with fields " +
+               "are preserved during the reordering.";
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(2);
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+            @Override
+            public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext ctx) {
+                J.ClassDeclaration cd = super.visitClassDeclaration(classDecl, ctx);
+
+                List<Statement> statements = cd.getBody().getStatements();
+                if (statements.isEmpty()) {
+                    return cd;
+                }
+
+                List<Statement> fieldDeclarations = new ArrayList<>();
+                List<Statement> otherStatements = new ArrayList<>();
+
+                // Separate field declarations from other statements
+                for (Statement statement : statements) {
+                    if (statement instanceof J.VariableDeclarations) {
+                        fieldDeclarations.add(statement);
+                    } else {
+                        otherStatements.add(statement);
+                    }
+                }
+
+                // If all fields are already at the top, no changes needed
+                if (fieldDeclarations.isEmpty() || isAlreadyOrdered(statements, fieldDeclarations)) {
+                    return cd;
+                }
+
+                // Reorder: fields first, then other statements
+                List<Statement> reorderedStatements = new ArrayList<>();
+
+                // Add field declarations with preserved comments and proper spacing
+                for (int i = 0; i < fieldDeclarations.size(); i++) {
+                    Statement field = fieldDeclarations.get(i);
+                    if (i == 0) {
+                        // First field should have the same prefix as the first statement originally had,
+                        // but preserve its original comments
+                        Space originalPrefix = field.getPrefix();
+                        Space firstStatementPrefix = statements.get(0).getPrefix();
+                        // Combine: use first statement's whitespace but preserve field's comments
+                        Space newPrefix = firstStatementPrefix.withComments(originalPrefix.getComments());
+                        field = field.withPrefix(newPrefix);
+                    }
+                    reorderedStatements.add(field);
+                }
+
+                // Add other statements
+                reorderedStatements.addAll(otherStatements);
+
+                return autoFormat(cd.withBody(cd.getBody().withStatements(reorderedStatements)), ctx);
```

Comment:
do we really need the autoformat here? I feel that might cause unexpected formatting on inconsistent projects.